syft/pkg/cataloger/common/cpe/java.go

```diff
@@ -241,11 +274,15 @@ func addGroupIDsFromGroupIDsAndArtifactID(groupID, artifactID string) (groupIDs
 	return groupIDs
 }
 
-func groupIDsFromJavaManifest(manifest *pkg.JavaManifest) []string {
+func groupIDsFromJavaManifest(pkgName string, manifest *pkg.JavaManifest) []string {
 	if manifest == nil {
 		return nil
 	}
 
+	if groupID, ok := defaultArtifactIDToGroupID[pkgName]; ok {
```

Comment:
Should this be before the manifest check? Even if the manifest is nil we can query to see if we have a group id for the `pkgName` in the map

Reply:
Good call. Updated.